new-lamassu-admin/src/pages/Blacklist/Blacklist.js

```diff
@@ -0,0 +1,183 @@
+    refetchQueries: () => ['getData']
+  })
+
+  const classes = useStyles()
+
+  const blacklistData = R.path(['blacklist'])(blacklistResponse) ?? []
+  const availableCurrencies =
+    R.path(['cryptoCurrencies'], blacklistResponse) ?? []
+
+  const formattedData = groupByCode(blacklistData)
+
+  const complianceConfig =
+    configData?.config && fromNamespace('compliance')(configData.config)
+
+  const rejectAddressReuse = complianceConfig?.rejectAddressReuse ?? false
+
+  const addressReuseSave = rawConfig => {
+    const config = toNamespace('compliance')(rawConfig)
+    return saveConfig({ variables: { config } })
+  }
+
+  const onClickSidebarItem = e => {
+    setClickedItem({ code: e.code, display: e.display })
+  }
+
+  const handleDeleteEntry = (cryptoCode, address) => {
+    deleteEntry({ variables: { cryptoCode, address } })
+  }
+
+  const isSelected = it => clickedItem.code === it.code
+
+  const toggleModal = () => setShowModal(!showModal)
+
+  const addToBlacklist = (cryptoCode, address) => {
+    addEntry({ variables: { cryptoCode, address } })
+  }
+
+  return (
+    <>
+      <TitleSection title="Blacklisted addresses">
+        <Link onClick={toggleModal}>Blacklist new addresses</Link>
+      </TitleSection>
+      <Grid container className={classes.grid}>
+        <Sidebar
+          data={availableCurrencies}
+          isSelected={isSelected}
+          displayName={it => it.display}
+          onClick={onClickSidebarItem}
+        />
+        <div className={classes.content}>
+          <Box display="flex" justifyContent="space-between" mb={3}>
+            <H4 noMargin className={classes.subtitle}>
+              {clickedItem.display
+                ? `${clickedItem.display} blacklisted addresses`
+                : ''}{' '}
+            </H4>
+            <Box
+              display="flex"
+              alignItems="center"
+              justifyContent="end"
+              mr="-5px">
+              <P>Reject reused addresses</P>
+              <Switch
+                checked={rejectAddressReuse}
+                onChange={event => {
+                  addressReuseSave({ rejectAddressReuse: event.target.checked })
+                }}
+                value={rejectAddressReuse}
+              />
+              <Label2>{rejectAddressReuse ? 'On' : 'Off'}</Label2>
+              <Tooltip width={304}>
+                <P>
+                  The "Reject reused addresses" option means that all addresses
+                  that are used once will be automatically rejected if there's
+                  an attempt to use them again on a new transaction.
+                </P>
+              </Tooltip>
+            </Box>
+          </Box>
+          <BlacklistTable
+            data={formattedData}
+            selectedCoin={clickedItem}
+            handleDeleteEntry={handleDeleteEntry}
+          />
+        </div>
+      </Grid>
+      <BlackListModal
+        showModal={showModal}
+        toggleModal={toggleModal}
```

Comment:
Name this onClose or similar, we don't need to ever toggle the modal inside the modal, pass down the `() => setShowModal(false)` to this